streams/src/test/java/org/apache/kafka/streams/integration/InternalTopicIntegrationTest.java

```diff
@@ -197,9 +199,13 @@ public void shouldCompactAndDeleteTopicsForWindowStoreChangelogs() throws Except
                 return Arrays.asList(value.toLowerCase(Locale.getDefault()).split("\\W+"));
             }
         })
-                .groupBy(MockMapper.<String, String>selectValueMapper())
-                .windowedBy(TimeWindows.of(1000).until(2000))
-                .count(Materialized.<String, Long, WindowStore<org.apache.kafka.common.utils.Bytes, byte[]>>as("CountWindows"));
+            .groupBy(MockMapper.<String, String>selectValueMapper())
+            .windowedBy(TimeWindows.of(1000).grace(Duration.ZERO))
```

Comment:
As a consequence of the checks I've added (that retention >= size+grace), we have to explicitly set grace period now if we want a retention less than 24h (the default grace period).